### PR TITLE
Cli to operate on dataset indices

### PR DESCRIPTION
**Pull request by eddyxu**
```
Usage: lq index [OPTIONS] <ACTION> <URI>

Arguments:
  <ACTION>  Actions on index [possible values: create]
  <URI>     Dataset URI

Options:
  -c, --column <NAME>          The column to build index on
  -n, --name <NAME>            Index name
  -t, --type <TYPE>            Set index type [possible values: ivf-pq]
  -p, --num-partitions <NUM>   Nunber of IVF partitions. Only useful when the index type is 'ivf-pq' [default: 64]
  -s, --num-sub-vectors <NUM>  Number of sub-vectors in Product Quantizer [default: 8]
  -h, --help                   Print help

```

### CODE REVIEW

File: rust/Cargo.toml [hunk 1]
```diff
@@ -46,6 +46,8 @@ url = "2.3"
 rand = "0.8.3"
 futures = "0.3"
 uuid = { version = "1.2", features = ["v4"] }
+faiss = {version = "0.11.0", features=["gpu"], optional = true }
```
Comment (changhiskhan) [inline rust/Cargo.toml:49]: Does this make the binary much larger?

File: rust/src/dataset.rs [hunk 2]
```diff
@@ -121,6 +123,7 @@ impl Dataset {
             Err(object_store::Error::NotFound { path: _, source: _ }) => { /* we are good */ }
             Err(e) => return Err(Error::from(e)),
         }
+        println!("After check manifest path exist");
```
Comment (changhiskhan) [inline rust/src/dataset.rs:126]: Remove prints?

**Review by changhiskhan — APPROVED** [review]
lfg how long does it take

File: rust/Cargo.toml (hunk 1 above)
Comment (eddyxu) [reply rust/Cargo.toml:49]: this needs to be able to locate `faiss_c.so` somewhere, so it does not make the binary larger, but crash if `faiss.so` not found.

**Pull request merged**